From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andy Boxer <andyboxer@gmail.com>
Date: Sat, 4 Jul 2026 13:47:34 +0300
Subject: [PATCH] bcm47xx: flush cache when duplicating mm

Flush the cache when duplicating an address space on legacy SSB-only
BCM47xx systems to avoid userspace cache-alias failures.

Signed-off-by: Andy Boxer <andyboxer@gmail.com>
---
--- a/arch/mips/include/asm/cacheflush.h
+++ b/arch/mips/include/asm/cacheflush.h
@@ -46,7 +46,11 @@
 extern void (*flush_cache_all)(void);
 extern void (*__flush_cache_all)(void);
 extern void (*flush_cache_mm)(struct mm_struct *mm);
+#if defined(CONFIG_BCM47XX_SSB) && !defined(CONFIG_BCM47XX_BCMA)
+#define flush_cache_dup_mm(mm)	flush_cache_mm(mm)
+#else
 #define flush_cache_dup_mm(mm)	do { (void) (mm); } while (0)
+#endif
 extern void (*flush_cache_range)(struct vm_area_struct *vma,
 	unsigned long start, unsigned long end);
 extern void (*flush_cache_page)(struct vm_area_struct *vma, unsigned long page, unsigned long pfn);
